client/ayon_maya/plugins/load/load_maya_usd_add_maya_reference.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+import contextlib
+
+from ayon_core.pipeline import load
+from ayon_maya.api.usdlib import (
+    containerise_prim,
+    iter_ufe_usd_selection
+)
+
+from maya import cmds
+import mayaUsd
+
+
+@contextlib.contextmanager
+def no_edit_mode(prim, restore_after=True):
+    """Ensure MayaReference prim is not in edit mode during context"""
+    pulled_node = mayaUsd.lib.PrimUpdaterManager.readPullInformation(prim)
+    ufe_path = None
+    try:
+        # remove edit state if pulled
+        if pulled_node:
+            import mayaUsdUtils
+            assert mayaUsdUtils.isPulledMayaReference(pulled_node)
+            cmds.mayaUsdDiscardEdits(pulled_node)
+
+            # Discarding the edits directly selects the prim
+            # so we can get the UFE path from selection
+            ufe_path = cmds.ls(selection=True, ufeObjects=True, long=True)[0]
+
+        yield prim, ufe_path, pulled_node
+    finally:
+        if restore_after and pulled_node and ufe_path:
+            cmds.mayaUsdEditAsMaya(ufe_path)
+
+
+class MayaUsdProxyAddMayaReferenceLoader(load.LoaderPlugin):
+    """Read USD data in a Maya USD Proxy
+
+    TODO: It'd be much easier if this loader would be capable of returning the
+        available containers in the scene based on the AYON URLs inside a USD
+        stage. That way we could potentially avoid the need for custom metadata
+        keys, stay closer to USD native data and rely solely on the
+        AYON:asset=blue,subset=modelMain,version=1 url
+
+    """
+
+    product_types = {"*"}
+    representations = ["*"]
+    extensions = ["ma", "mb"]
+
+    label = "USD Add Maya Reference"
+    order = 1
+    icon = "code-fork"
+    color = "orange"
+
+    identifier_key = "ayon_identifier"
+
+    def load(self, context, name=None, namespace=None, options=None):
+
+        selection = list(iter_ufe_usd_selection())
+        assert len(selection) == 1, "Select only one PRIM please"
```

Comment:
Does it mean it is only allowed one geometry in the loaded asset(or can we load with multiple assets)? Maybe we can exclude some families to load this as we can load stuff via this loader if it is also layout product(which is mostly with multiple assets). 
![image](https://github.com/user-attachments/assets/300790e0-3ac7-4b2f-81a5-0fe65691f466)

Reply:
The error means that you need to SELECT a PRIM inside the Maya USD Proxy to 'reference into'. Unfortunately since it 'references into a prim' there isn't really something more intuitive I could think of whilst still allowing to load anywhere inside the USD hierarchy.